src/System/Net/CertificateHelper.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+using System.Collections.Concurrent;
+using System.Net;
+
+namespace SpaceWizards.HttpListener
+{
+    public class CertificateHelper
+    {
+        private static readonly string[] tlds = {
```

Comment:
This is an extremely outdated list of TLDs. You cannot hardcode something like this.